macros/src/cbor_serializable.rs

```diff
@@ -0,0 +1,420 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse_macro_input, Attribute, Data, DeriveInput, Fields, Index, Lit, Meta, MetaList,
+    NestedMeta, Type,
+};
+
+pub(crate) fn cbor_serializable(input: TokenStream) -> TokenStream {
```

Comment:
I'm a big fan of implementing custom derive macros, but reimplementing `serde` seems a bit much :D
I think you can make the `CborSerializable` derive macro rely on the `Serialize` implementation.
```
impl AsCborValue for Type {
	fn from_cbor_value(value: Value) -> Result<Self> {
		value.deserialized().map_err(serde::de::Error::custom)
	}

	fn to_cbor_value(self) -> Result<Value> {
		Value::serialized(&self)
	}
}
```

Reply:
`coset` objects don't implement `Serialize`/`Deserialize` but you can use a custom de/serializer based on `AsCborValue` for them if necessary.

Reply:
As before, I tested this but it doesn't handle tagged well, Strange is it serialize as tagged but it looses the tag on unserialize.

Reply:
migrated to serialize using serde and coset/cuborium